avalon/tools/widgets.py

```diff
@@ -129,204 +101,68 @@ def get_active_index(self):
         return self.view.currentIndex()
 
     def get_selected_assets(self):
-        """Return the assets' ids that are selected."""
+        """Return the documents of selected assets."""
         selection = self.view.selectionModel()
         rows = selection.selectedRows()
-        return [row.data(self.model.ObjectIdRole) for row in rows]
+        assets = [row.data(self.model.DocumentRole) for row in rows]
 
-    def set_silo(self, silo):
-        """Set the active silo tab"""
-        self.silo.set_current_silo(silo)
+        # NOTE: skip None object assumed they are silo (backwards comp.)
+        return [asset for asset in assets if asset]
 
-    def select_assets(self, assets, expand=True):
-        """Select assets by name.
+    def select_assets(self, assets, expand=True, key="name"):
+        """Select assets by item key.
 
         Args:
-            assets (list): List of asset names
+            assets (list): List of asset values that can be found under
+                specified `key`
             expand (bool): Whether to also expand to the asset in the view
+            key (string): Key that specifies where to look for `assets` values
 
         Returns:
             None
 
+        Default `key` is "name" in that case `assets` should contain single
+        asset name or list of asset names. (It is good idea to use "_id" key
+        instead of name in that case `assets` must contain `ObjectId` object/s)
+        It is expected that each value in `assets` will be found only once.
+        If the filters according to the `key` and `assets` correspond to
+        the more asset, only the first found will be selected.
```

Comment:
Seems to have some odd grammar, maybe this is better?
```
        Default `key` is "name" in that case `assets` should contain single
        asset name or list of asset names. It is recommended to use "_id" key
        instead of name. In that case `assets` must contain `ObjectId` objects.
        It is assumed that each value in `assets` is found only once.
        On multiple matches only the first found will be selected.
```